src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = 'final Street Edit';
+            contactsAddrs.contacts[i].MailingCity = 'final City Edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assert(contact.MailingStreet.contains('final Street Edit'));
+            system.assert(contact.MailingCity.contains('final City Edit'));
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+ 
+         // verify the previous addresses got Default cleared.
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c from Address__c];
+        // each set of edits created a new address object 
+        system.assertEquals(16, ListAddr.size());
+        for (Address__c addr : ListAddr) {
+            boolean fNewDefault = (addr.MailingStreet__c.contains('final Street Edit'));
+            system.assertEquals(fNewDefault, addr.Default_Address__c);
+        }                
+    }
+
+    /*********************************************************************************************************
+    @description
+        update Address' Mailing address and mark it as override, to cause creation of new non-default Address object  
+    verify:
+        new Address created
+        original Addresses still default    
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateAddrMarkOverride() {
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        // modify the Contact addresses directly, and set override
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = 'Direct Street Edit';
+            contactsAddrs.contacts[i].MailingCity = 'Direct City Edit';
+            contactsAddrs.contacts[i].is_Address_Override__c = true;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause or contain the number of records redurned, or adding a `LIMIT` clause to cap the maximum number of records returned.